src/ppom-controller.ts

```diff
@@ -337,25 +347,41 @@ export class PPOMController extends BaseControllerV2<
    * This function receives a callback that will be called with the PPOM.
    *
    * @param callback - Callback to be invoked with PPOM.
+   * @param networkClientId - Optional network client ID to use for the PPOM requests.
    */
   async usePPOM<T>(
-    callback: (ppom: any) => Promise<T>,
+    callback: (ppom: unknown) => Promise<T>,
+    networkClientId?: NetworkClientId,
   ): Promise<T & { providerRequestsCount: Record<string, number> }> {
     if (!this.#securityAlertsEnabled) {
       throw Error('User has securityAlertsEnabled set to false');
     }
-    if (!this.#networkIsSupported(this.#chainId)) {
-      throw Error('Blockaid validation is available only on ethereum mainnet');
+
+    let chainId = this.#chainId;
+    let provider = this.#provider;
+    if (networkClientId) {
+      const networkClient = this.messagingSystem.call(
+        'NetworkController:getNetworkClientById',
+        networkClientId,
+      );
+      chainId = networkClient.configuration.chainId;
+      provider = networkClient.provider;
     }
 
-    await this.#getNewFilesForCurrentChainIfNeeded();
+    if (!this.#networkIsSupported(chainId)) {
+      throw Error('Blockaid validation is available only on ethereum mainnet');
+    }
 
-    this.#providerRequests = 0;
-    this.#providerRequestsCount = {};
+    await Promise.all([
+      this.#getNewFilesForChainIfNeeded(chainId),
+      this.#initialisePPOM(),
+    ]);
 
-    await this.#initialisePPOM();
```

Comment:
Above line is required for mobile as initialisation in constructor for some reason does not works in react native.